src/components/Post.js

```diff
@@ -72,22 +111,68 @@ class Post extends React.Component {
       </LazyLoader>}
       <Details {...{expanded, onExpand}}/>
       {linkPreview && <LinkPreview {...{linkPreview}}/>}
-      <div className='voting post-section'><VoteButton/><Voters/></div>
+      <div className='voting post-section'>
+        <VoteButton/>
+        <Voters/>
+      </div>
       <Attachments/>
+      {hasFeature(currentUser, CONTRIBUTORS) && isCompleteRequest &&
+        <div className='request-completed-bar'>
+          <div className='request-complete-heading'>
+            <input className='toggle'
+              type='checkbox'
+              checked={!!post.fulfilled_at}
+              readOnly={!canEdit} />
+            <Contributors />
+          </div>
+        </div>
+      }
       <CommentSection {...{post, expanded, onExpand, comments}}/>
+      {hasFeature(currentUser, CONTRIBUTORS) && isIncompleteRequest &&
+        <div className='request-completed-bar'>
+          <div className='request-complete-heading'>
+            <input type='checkbox'
+              className='toggle'
+              checked={this.state.requestCompleting}
+              onChange={toggleRequestCompleting} />
+            <p className='request-complete-message'>
+              {this.state.requestCompleting ?
+                'Click the checkmark if your request has been completed!' :
+                'Awesome! Who helped you?'}
+            </p>
+          </div>
+          {this.state.requestCompleting &&
+            <div className='buttons'>
+              <a className='cancel' onClick={toggleRequestCompleting}>
+                <span className='icon icon-Fail'></span>
+              </a>
+              <TagInput className='request-complete-people-input'
+                choices={contributorChoices}
+                handleInput={contributorHandleInput}
+                onSelect={addContributor}
+                onRemove={removeContributor}
+                tags={contributors} />
+              <a className='done' onClick={completeRequest}>
+                Done
+              </a>
+            </div>
+          }
+        </div>
+      }
     </div>
   }
 }
 
-export default compose(
-  connect((state, { post }) => {
-    return {
-      comments: getComments(post, state),
-      community: getCurrentCommunity(state),
-      post: denormalizedPost(post, state)
-    }
-  })
-)(Post)
+export default compose(connect((state, {post}) => {
+  return {
+    comments: getComments(post, state),
+    community: getCurrentCommunity(state),
+    post: denormalizedPost(post, state),
+    contributorChoices: reject(
+      state.typeaheadMatches.invite, {id: post.user_id}
```

Comment:
if you want to avoid having to stub `typeaheadMatches` in tests, you can do `get('invite', state.typeaheadMatches)` instead of `state.typeaheadMatches.invite` here. but even if you leave this as-is, you only need to stub `typeaheadMatches` itself in the tests; it doesn't need any content.